dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizer.scala

```diff
@@ -461,10 +461,10 @@ object DistriOptimizer {
     })
   }
 
-
-  private def getModel[T: ClassTag](models: RDD[Cache[T]],
-      parameters: AllReduceParameter[T])
-  : Module[T] = {
+  private def getModel[T: ClassTag](
+      models: RDD[Cache[T]],
+      parameters: AllReduceParameter[T],
+      result: Module[T] = null): Module[T] = {
```

Comment:
There is no need to change the interface; you can unconditionally copy the model you get in this method to the user provided model (Optimizer.model), either at the end of this method or at the end of the optimize() method.